src/System.Data.SqlClient/tests/ManualTests/SQL/SqlNamedPipesTest/SqlNamedPipesTest.cs

```diff
@@ -15,51 +15,43 @@ public static void ValidConnStringTest()
             SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(DataTestUtility.NpConnStr);
             builder.ConnectTimeout = 5;
 
-            string plainConnString = builder.ConnectionString;
-            builder.DataSource = "np:" + GetHostFromDataSource(builder.DataSource);
-            string serverNameOnlyConnString = builder.ConnectionString;
-
             if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                OpenBadConnection<PlatformNotSupportedException>(plainConnString);
-                OpenBadConnection<PlatformNotSupportedException>(serverNameOnlyConnString);
+                OpenBadConnection<PlatformNotSupportedException>(builder.ConnectionString);
             }
             else
             {
-                OpenGoodConnection(plainConnString);
-                OpenGoodConnection(serverNameOnlyConnString);
+                OpenGoodConnection(builder.ConnectionString);
             }
         }
 
 #if MANAGED_SNI
         [CheckConnStrSetupFact]
         public static void InvalidConnStringTest()
         {
-            const string invalidConnStringError = "A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: Named Pipes Provider, error: 25 - Connection string is not valid)";
+            string invalidConnStringError = SystemDataResourceManager.Instance.SNI_ERROR_25;
 
             SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(DataTestUtility.NpConnStr);
             builder.ConnectTimeout = 2;
 
-            string host = GetHostFromDataSource(builder.DataSource);
-
             // Using forward slashes
-            builder.DataSource = "np://" + host + "/pipe/sql/query";
+            builder.DataSource = "np://NotARealServer/pipe/sql/query";
```

Comment:
Instead of `NotARealServer`, which actually could be a real server on someone's network, how about something like `Guid.NewGuid().ToString("N")`?